File: frontend/src/api/tutor.ts
import api from './axios'
import { type TaskType } from './tasks'

export type EvaluationResult = 'PASS' | 'FAIL'
export type ChatStatus = 'PENDING' | 'PASS' | 'FAIL'

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface TutorEvaluationRequest {
  wordId: string
  taskType: TaskType
  userResponse: string
  chatId?: string | null
}

export interface TutorEvaluationResponse {
  result: EvaluationResult
  feedback: string
  hint: string | null
  answerRevealed: boolean
  chatId: string | null
  expectedAnswer: string | null
  reason: string | null
}

export interface ChatContinueRequest {
  message: string
}

export interface TutorChatResponse {
  id: string
  userId: string
  wordId: string
  taskType: TaskType
  messages: ChatMessage[]
  finalResult: ChatStatus
  createdAt: string
}

export interface ChatListItem {
  id: string
  wordId: string
  word: string
  meaning: string
  taskType: TaskType
  finalResult: ChatStatus
  createdAt: string
  messageCount: number
}

export const tutorApi = {
  evaluateResponse: async (data: TutorEvaluationRequest): Promise<TutorEvaluationResponse> => {
    const response = await api.post<TutorEvaluationResponse>('/api/tutor/evaluate', data)
    return response.data
  },

  continueChat: async (chatId: string, message: string): Promise<TutorEvaluationResponse> => {
    const response = await api.post<TutorEvaluationResponse>(`/api/tutor/chat/${chatId}`, {
      message
    })
    return response.data
  },

  getChatHistory: async (chatId: string): Promise<TutorChatResponse> => {
    const response = await api.get<TutorChatResponse>(`/api/tutor/chat/${chatId}`)
    return response.data
  },

  listChats: async (limit: number = 50, offset: number = 0): Promise<ChatListItem[]> => {
    const response = await api.get<ChatListItem[]>('/api/tutor/chats', {
      params: { limit, offset }
    })
    return response.data
  },
}

File: backend/app/routers/tutor.py
from fastapi import APIRouter, Depends, HTTPException, status, Query
from typing import List
from app.models.tutor_chat import (
    TutorEvaluationRequest, 
    TutorEvaluationResponse,
    ChatContinueRequest,
    TutorChatResponse,
    ChatListItem
)
from app.middleware.auth_middleware import get_current_user
from app.services.tutor_service import evaluate_response, continue_chat, get_chat_history, list_user_chats

router = APIRouter(prefix="/api/tutor", tags=["Tutor"])

@router.post("/evaluate", response_model=TutorEvaluationResponse)
async def evaluate_endpoint(
    request: TutorEvaluationRequest,
    current_user: dict = Depends(get_current_user)
):
    """Evaluate user response using AI tutor"""
    try:
        return await evaluate_response(
            current_user["user_id"], 
            request,
            chat_id=request.chatId
        )
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=str(e)
        )

@router.post("/chat/{chat_id}", response_model=TutorEvaluationResponse)
async def continue_chat_endpoint(
    chat_id: str,
    request: ChatContinueRequest,
    current_user: dict = Depends(get_current_user)
):
    """Continue a chat conversation"""
    try:
        return await continue_chat(
            current_user["user_id"],
            chat_id,
            request.message
        )
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=str(e)
        )

@router.get("/chat/{chat_id}", response_model=TutorChatResponse)
async def get_chat_endpoint(
    chat_id: str,
    current_user: dict = Depends(get_current_user)
):
    """Get chat history"""
    try:
        return await get_chat_history(
            current_user["user_id"],
            chat_id
        )
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=str(e)
        )

@router.get("/chats", response_model=List[ChatListItem])
async def list_chats_endpoint(
    limit: int = Query(50, ge=1, le=100),
    offset: int = Query(0, ge=0),
    current_user: dict = Depends(get_current_user)
):
    """List all chats for the current user"""
    try:
        return await list_user_chats(
            current_user["user_id"],
            limit,
            offset
        )
    except Exception as e:
        raise HTTPException(
            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
            detail=str(e)
        )
